src/txmempool.h

```diff
@@ -62,6 +68,27 @@ class CTxMemPoolEntry
     size_t DynamicMemoryUsage() const { return nUsageSize; }
 };
 
+// extracts a TxMemPoolEntry's transaction hash
+struct mempoolentry_txid
+{
+    typedef uint256 result_type;
+    result_type operator() (const CTxMemPoolEntry &entry) const
+    {
+        return entry.GetTx().GetHash();
+    }
+};
+
+class CompareTxMemPoolEntryByFee
```

Comment:
If we're only having 1 comparator (and having more than one would be over-complicating this), simply CompareTxMemPoolEntry or TxMemPoolEntryComparator is enough.
Of course, both suggestions are just bike-shedding.